docs: add FAQ for unrelated PR changes (#6415)
stdlib-js/stdlib: docs/contributing/FAQ.md

To see other available bot commands, comment `/stdlib help` on your PR.

<a name="auto-push"></a>

## Why were many unrelated files automatically pushed to my PR when I committed my changes?

When you open a pull request or push changes to your feature branch, GitHub compares your feature branch against your `develop` branch and shows all the differences. If your feature branch contains outdated or extra changes, they will appear in the PR, even if they are unrelated to your work.

> To fix this, ensure that your feature branch is based on your latest `develop` branch. You can do this by merging your `develop` branch into your feature branch:

Reviewer: Another thing you can do is to do `git pull upstream develop`. Right now, you are assuming that your origin `develop` is in sync. That is likely not going to be the case for many new contributors who are not constantly syncing their local develop with the upstream.

Reviewer: I might amend the flow to be

```bash
$ git checkout develop
$ git pull upstream develop
$ git push origin develop
$ git checkout feature-branch
$ git merge develop
```

```bash
$ git checkout develop
$ git pull origin develop
$ git checkout feature-branch
$ git merge develop
```

After merging, push your changes to the remote repository:

```bash
$ git push origin feature-branch # git push also works
```

> > **Note**: At stdlib, we recommend using `merge` instead of `rebase` once a PR is open. Rebasing rewrites your branch history, which usually requires a force-push to update the remote branch. This can disrupt other contributors who are reviewing or collaborating on your PR. Since stdlib uses squash and merge for PRs, we don’t require a clean, linear commit history. Merge commits are totally fine as long as your diff only contains relevant changes. If you want to learn more about rebasing/merging, you can refer to our [Git guide][git-guide].

Reviewer: ```suggestion
> **Note**: When developing stdlib, we recommend using `merge` instead of `rebase` once a PR is open. Rebasing rewrites your branch history, which usually requires a force-push to update the remote branch. This can disrupt other contributors who are reviewing or collaborating on your PR. Since stdlib uses squash and merge for PRs, we don't require a clean, linear commit history. Merge commits are totally fine as long as your diff only contains relevant changes. If you want to learn more about rebasing/merging, you can refer to our [Git guide][git-guide].
```

You may want to configure your editor's font to not use fancy quotes.

Author: :(

To see other available bot commands, comment `/stdlib help` on your PR.

<a name="auto-push"></a>

## Why were many unrelated files automatically pushed to my PR when I committed my changes?

When you open a pull request or push changes to your feature branch, GitHub compares your feature branch against your `develop` branch and shows all the differences. If your feature branch contains outdated or extra changes, they will appear in the PR, even if they are unrelated to your work.

To fix this, ensure that your feature branch is based on the latest `develop` branch. You can do this by updating your local `develop` branch and then merging it into your feature branch:

```bash
$ git checkout develop
$ git pull upstream develop
$ git push origin develop
$ git checkout feature-branch
$ git merge develop
```

After merging, push your changes to the remote repository:

```bash
$ git push origin feature-branch # git push also works
```

> > **Note**: When developing stdlib, we recommend using `merge` instead of `rebase` once a PR is open. Rebasing rewrites your branch history, which usually requires a force-push to update the remote branch. This can disrupt other contributors who are reviewing or collaborating on your PR. Since stdlib uses squash and merge for PRs, we don't require a clean, linear commit history. Merge commits are totally fine as long as your diff only contains relevant changes. If you want to learn more about rebasing/merging, you can refer to our [Git guide][git-guide].

Reviewer: ```suggestion
> **Note**: When developing stdlib, we recommend using `merge` instead of `rebase` once a PR is open. Rebasing rewrites your branch history, which usually requires a force-push to update the remote branch. This can disrupt other contributors who are reviewing or collaborating on your PR. Since stdlib uses squash and merge for PRs, we don't require a clean, linear commit history. Merge commits are acceptable as long as your diff only contains relevant changes. If you want to learn more about rebasing/merging, you can refer to our [Git guide][git-guide].
```
